documentation/resource_machine_name.md

```diff
@@ -12,7 +12,7 @@ A `dns_domain` property can be optionally specified. This will be tacked on to t
 end of the specified `hostname` property to form a fully-qualified domain name
 that the system `HostName` will be set to.
 
-When the state of a `machine_name` resource changes, an `ohai` resource is notified
+When the state of the `machine_name` resource changes, an `ohai` resource is notified
```

Comment:
@v-anshie The reason we have "the" and "a" used seemingly interchangeably is the difference between a [class and an object/instance](https://en.wikipedia.org/wiki/Instance_(computer_science). A single blueprint can be responsible for the construction of many houses, each one maybe a little different, depending on different *properties*. We should revisit some of these changes to make sure we're using the right language.

Reply:
```suggestion
When the state of a `machine_name` resource changes, an `ohai` resource is notified
```